packages/jest-core/src/runGlobalHook.ts

```diff
@@ -85,8 +56,6 @@ export default async ({
       }
 
       await globalModule(globalConfig);
-
-      revertHook();
```

Comment:
Since we previously waited with removing the hook, lazy requires would still be transpiled. Do you think it makes sense to allow passing a callback which would be `await`ed before removing the require hook?

So e.g.
```ts
requireAndTranspileModule(moduleName: string, cb?: (module: any) => Promise<void>): Promise<any>
```
And we'd `await cb(module)` if `cb` was passed

Reply:
I'm happy to make this change myself, btw

Reply:
Wouldn't `requireAndTranspileModule` need an await when ever it is called?

Reply:
Yes. I think that's fine? Or we could add a separate function. Or even change the return type depending on whether the cb is passed or not

Reply:
Currently 
`jest-core > runGlobalHooks.ts` is async already so no problem there
`jest-runner > runTest.ts` which will also use this is async
`jest-runtime > cli/index.ts` it will be used within a callback which is currently sync

Except for one case the others are async already

Reply:
That can probably be made async as well. As it's a CLI it shouldn't be a breaking change, should it?

Reply:
Ok, I'll add it tommorow